asyncio warning

```
src/base/system.c:468:3: warning: redefinition of typedef 'ASYNCIO' is a C11 feature [-Wtypedef-redefinition]
} ASYNCIO;
  ^
src/base/system.h:370:24: note: previous definition is here
typedef struct ASYNCIO ASYNCIO;
                       ^
1 warning generated.

```

--- a/src/base/system.c
+++ b/src/base/system.c
@@ -448,7 +448,7 @@ int io_flush(IOHANDLE io)
 
 #define ASYNC_BUFSIZE 8 * 1024
 
-typedef struct ASYNCIO
+struct ASYNCIO
 {
 	LOCK lock;
 	IOHANDLE io;
@@ -465,7 +465,7 @@ typedef struct ASYNCIO
 	int error;
 	unsigned char finish;
 	unsigned char refcount;
-} ASYNCIO;
+};
 
 enum
 {

